protocol/contracts/amo/helpers/BalancerPoolHelper.sol

```diff
@@ -0,0 +1,428 @@
+        protocolTokenAmount = quoteTokenBalanceInLP == 0
+            ? quoteTokenAmount
+            : (protocolTokenBalanceInLP * quoteTokenAmount / quoteTokenBalanceInLP);
+
+        requestData.assets = new IERC20[](2);
+        requestData.maxAmountsIn = new uint256[](2);
+        
+        (requestData.assets[0], requestData.assets[1]) = protocolTokenIndexInBalancerPool == 0 ? (protocolToken, quoteToken) : (quoteToken, protocolToken);
+        (requestData.maxAmountsIn[0], requestData.maxAmountsIn[1]) = protocolTokenIndexInBalancerPool == 0 ? (protocolTokenAmount, quoteTokenAmount) : (quoteTokenAmount, protocolTokenAmount);
+        //uint256 joinKind = 1; //EXACT_TOKENS_IN_FOR_BPT_OUT
+        bytes memory encodedUserdata = abi.encode(uint256(1), requestData.maxAmountsIn, 0);
+        requestData.userData = encodedUserdata;
+        requestData.fromInternalBalance = false;
+
+        (expectedBptAmount, ) = balancerHelpers.queryJoin(balancerPoolId, amo, amo, requestData);
+        minBptAmount = applySlippage(expectedBptAmount, slippageBps);
+
+        // update `requestData` with the `minBptAmount` to which the slippage was applied
+        encodedUserdata = abi.encode(uint256(1), requestData.maxAmountsIn, minBptAmount);
+        requestData.userData = encodedUserdata;
+    }
+
+    /// @notice Get the quote used to remove liquidity
+    /// @dev Since this is not the view function, this should be called with `callStatic`
+    function proportionalRemoveLiquidityQuote(
+        uint256 bptAmount,
+        uint256 slippageBps
+    ) external override returns (
+        uint256 expectedProtocolTokenAmount,
+        uint256 expectedQuoteTokensAmount,
+        uint256 minProtocolTokenAmount,
+        uint256 minQuoteTokensAmount,
+        IBalancerVault.ExitPoolRequest memory requestData
+    ) {
+        requestData.assets = new address[](2);
+        requestData.minAmountsOut = new uint256[](2);
+
+        (requestData.assets[0], requestData.assets[1]) = protocolTokenIndexInBalancerPool == 0
+            ? (address(protocolToken), address(quoteToken))
+            : (address(quoteToken), address(protocolToken));
+        // EXACT_BPT_IN_FOR_TOKENS_OUT index is 1 for exitKind
+        bytes memory encodedUserdata = abi.encode(uint256(1), bptAmount);
+        requestData.userData = encodedUserdata;
+        requestData.toInternalBalance = false;
+
+        (, requestData.minAmountsOut) = balancerHelpers.queryExit(balancerPoolId, amo, amo, requestData);
+        (expectedProtocolTokenAmount, expectedQuoteTokensAmount) = protocolTokenIndexInBalancerPool == 0
+            ? (requestData.minAmountsOut[0], requestData.minAmountsOut[1])
+            : (requestData.minAmountsOut[1], requestData.minAmountsOut[0]);
+        minProtocolTokenAmount = applySlippage(expectedProtocolTokenAmount, slippageBps);
```

Comment:
nit: have in mind if slippage is large enough, could lead to frontrunning. perhaps a way to control with maxSlippage?

Reply:
same

Reply:
Decided not to implement - responsibility of the bot